packages/@aws-cdk/aws-msk/lib/cluster-props.ts

```diff
@@ -0,0 +1,340 @@
+   * Must have at least 2 subnets in two different AZs.
+   */
+  readonly vpc: ec2.IVpc;
+  /**
+   * Where to place the nodes within the VPC.
+   * Amazon MSK distributes the broker nodes evenly across the subnets that you specify.
+   * The subnets that you specify must be in distinct Availability Zones.
+   * Client subnets can't be in Availability Zone us-east-1e.
+   *
+   * @default - the Vpc default strategy if not specified.
+   */
+  readonly vpcSubnets?: ec2.SubnetSelection;
+  /**
+   * The EC2 instance type that you want Amazon MSK to use when it creates your brokers.
+   *
+   * @see https://docs.aws.amazon.com/msk/latest/developerguide/msk-create-cluster.html#broker-instance-types
+   * @default kafka.m5.large
+   */
+  readonly instanceType?: ec2.InstanceType;
+  /**
+   * The AWS security groups to associate with the elastic network interfaces in order to specify who can
+   * connect to and communicate with the Amazon MSK cluster.
+   *
+   * @default - create new security group
+   */
+  readonly securityGroups?: ec2.ISecurityGroup[];
+  /**
+   * Information about storage volumes attached to MSK broker nodes.
+   *
+   * @default - 1000 GiB EBS volume
+   */
+  readonly storageInfo?: StorageInfo;
+  /**
+   * The distribution of broker nodes across Availability Zones. This is an optional parameter. If you don't
+   * specify it, Amazon MSK gives it the value DEFAULT.
+   * You can also explicitly set this parameter to the value DEFAULT.
+   *
+   * **No other values are currently allowed.**
+   *
+   * @default - DEFAULT
+   */
+  readonly brokerAzDistribution?: string;
+}
+
+/**
+ * Information about storage volumes attached to MSK broker nodes.
+ */
+export interface StorageInfo {
+  /**
+   * EBS volume information.
+   *
+   * @default - 1000 GiB volume size
+   */
+  readonly ebsStorageInfo?: EbsStorageInfo;
+}
+
+/**
+ * EBS volume information.
+ */
+export interface EbsStorageInfo {
+  /**
+   * The size in GiB of the EBS volume for the data drive on each broker node.
+   *
+   * @default 1000
+   */
+  readonly volumeSize?: number;
+  /**
+   * The AWS KMS key for encrypting data at rest.
+   *
+   * @default Uses AWS managed CMK (aws/kafka)
+   */
+  readonly kmsKey?: kms.IKey;
+}
+
+/**
+ * The Amazon MSK configuration to use for the cluster.
+ * Note: There is currently no Cloudformation Resource to create a Configuration
+ */
+export interface ClusterConfigurationInfo {
+  /**
+   * The Amazon Resource Name (ARN) of the MSK configuration to use.
+   * For example, arn:aws:kafka:us-east-1:123456789012:configuration/example-configuration-name/abcdabcd-1234-abcd-1234-abcd123e8e8e-1.
+   */
+  readonly arn: string;
+  /**
+   * The revision of the Amazon MSK configuration to use.
+   */
+  readonly revision: number;
+}
+
+/**
+ * The level of monitoring for the MSK cluster
+ *
+ * @see https://docs.aws.amazon.com/msk/latest/developerguide/monitoring.html#metrics-details
+ */
+export enum ClusterMonitoringLevel {
```

Comment:
I'm not sure if an enum is a good fit for this. This is the second time I've had to update this enum as a new monitoring level has been released (https://aws.amazon.com/about-aws/whats-new/2020/11/amazon-msk-offers-consumer-lag-metrics-select-topic-level-metrics-free/). Maybe should replace it with a class like [KafkaVersion](https://github.com/aws/aws-cdk/blob/ba75c8b45a517274cbc1874efce1ce1dd2c926f1/packages/%40aws-cdk/aws-msk/lib/cluster-version.ts) so that users can escape the enum with a custom value.